examples/with-validation/types/post.go

```diff
@@ -0,0 +1,18 @@
+package types
+
+import "time"
+
+type Post struct {
+	ID          int       `json:"id" binding:"required"`
+	Name        string    `json:"name" binding:"required"`
+	Body        string    `json:"body"`
+	PublishedAt time.Time `json:"published_at" binding:"required"`
+	Author      Author    `json:"author" binding:"required"`
+	Comments    []Comment `json:"comments" binding:"required,unique"`
+}
+
+type PostDTO struct {
```

Comment:
Didn't we discuss the behaviour as on request body `binding:required` indicates required, but on the response `omitempty` indicates optional (therefore required by default)?

Reply:
I was going to do that in a separate PR but I can add it here if you want, I think it might be slightly more nuanced so I wanted to have a bit of a chat and a think.

Reply:
Yeah ok, can you document this behaviour as a comment somewhere, just so we don't forget